packages/nextclade/src/sort/minimizer_search.rs

```diff
@@ -243,32 +243,40 @@ const fn invertible_hash(x: u64) -> u64 {
   x
 }
 
+const NUCLEOTIDE_LOOKUP: [(bool, u8); 256] = {
+  let mut table = [(false, 0); 256]; // Non-ACGT
+  table[b'A' as usize] = (true, 0b11); // A=11=3
+  table[b'a' as usize] = (true, 0b11);
+  table[b'T' as usize] = (true, 0b10); // T=10=2
+  table[b't' as usize] = (true, 0b10);
+  table[b'G' as usize] = (true, 0b00); // G=00=0
+  table[b'g' as usize] = (true, 0b00);
+  table[b'C' as usize] = (true, 0b01); // C=01=1
+  table[b'c' as usize] = (true, 0b01);
+  table
+};
+
 fn get_hash(kmer: &[u8], params: &MinimizerIndexParams) -> u64 {
   let cutoff = params.cutoff as u64;
 
   let mut x = 0;
   let mut j = 0;
 
-  for (i, nuc) in kmer.iter().enumerate() {
-    let nuc = *nuc as char;
-
+  // Create a bit-packed representation of the kmer
+  // where each nucleotide is represented by 2 bits:
+  // A=11, C=10, G=00, T=01
+  // Skip every third nucleotide to pick up conserved patterns
+  for (i, &nuc) in kmer.iter().enumerate() {
     if i % 3 == 2 {
-      continue; // skip every third nucleotide to pick up conserved patterns
-    }
-
-    if !"ACGT".contains(nuc) {
-      return cutoff + 1; // break out of loop, return hash above cutoff
-    }
-
-    // A=11=3, C=10=2, G=00=0, T=01=1
```

Comment:
This comment was actually wrong - T and C should have been interchanged.

Reply:
That's why the initial preview finds no matches (I used the comment rather than the code to implement the lookup table)